src/MCPClient/tests/test_create_dataverse_mets.py

```diff
@@ -0,0 +1,229 @@
+                              'ddi_count')
+
+dv_1 = DataverseMDIndex("4.8.6",
+                        "Test Dataset",
+                        "http://dx.doi.org/10.5072/FK2/XSAZXH",
+                        "doi",
+                        "web_ui_demo.dataverse.org.doi.10.5072.1.json",
+                        14, 3, 11, 1)
+dv_2 = DataverseMDIndex("4.8.6",
+                        "Bala Parental Alienation Study: Canada, United "
+                        "Kingdom, and Australia 1984-2012 [test]",
+                        "https://doi.org/10.5072/FK2/UNMEZF",
+                        "doi",
+                        "web_ui_demo.dataverse.org.doi.10.5072.2.json",
+                        47, 6, 41, 5)
+dv_3 = DataverseMDIndex("4.8.6",
+                        "A study with restricted data",
+                        "https://doi.org/10.5072/FK2/WZTJWN",
+                        "doi",
+                        "web_ui_demo.dataverse.org.doi.10.5072.3.json",
+                        5, 2, 3, 1)
+dv_4 = DataverseMDIndex("4.8.6",
+                        "A study of my afternoon drinks",
+                        "https://doi.org/10.5072/FK2/6PPJ6Y",
+                        "doi",
+                        "api_demo.dataverse.org.doi.10.5072.4.json",
+                        11, 3, 8, 2)
+dv_5 = DataverseMDIndex("4.8.6",
+                        "Botanical Test",
+                        "https://doi.org/10.5072/FK2/8KDUHM",
+                        "doi",
+                        "api_demo.dataverse.org.doi.10.5072.5.json",
+                        11, 2, 9, 1)
+dv_6 = DataverseMDIndex("4.8.6",
+                        "Research Data Management (RDM) Survey of Queen's "
+                        "University's Engineering and Science Departments",
+                        "https://hdl.handle.net/10864/11651",
+                        "hdl",
+                        "api_demo.dataverse.org.doi.10.5072.6.json",
+                        19, 3, 16, 2)
+dv_7 = DataverseMDIndex("4.8.6",
+                        "A study of my afternoon drinks",
+                        "https://doi.org/10.5072/FK2/6PPJ6Y",
+                        "doi",
+                        "api_demo.dataverse.org.doi.10.5072.7.json",
+                        11, 3, 8, 2)
+dv_8 = DataverseMDIndex("4.8.6",
+                        "Depress",
+                        "https://doi.org/10.5072/FK2/NNTESQ",
+                        "doi",
+                        "api_demo.dataverse.org.doi.10.5072.8.json",
+                        11, 3, 8, 2)
+
+
+class TestDataverseExample(object):
+    """Dataverse test runner class."""
+
+    write_dir = "fixtures/dataverse/dataverse_sources/dataverse_mets/"
+    fixture_path = "fixtures/dataverse/dataverse_sources"
+
+    THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+    FIXTURES_DIR = os.path.join(THIS_DIR, fixture_path)
+    WRITE_DIR = os.path.join(THIS_DIR, write_dir)
```

Comment:
The only thing that it's not obvious from the behaviour of `tmpdir` is that it does not return a string but a special kind of object, but you can convert it into a string to get its physical path.